tests/sagediff.py

```diff
@@ -217,105 +511,112 @@ def compare_catalogs(g1, g2):
         trange = xrange
 
     ignored_fields = [a for a in g1.ignored_fields if a in g2.ignored_fields]
-    for treenum in trange(g1.totntrees):
-        if g1.ngal_per_tree[treenum] == 0:
+
+    failed_fields = [] 
+
+    # Load all the galaxies in from all trees.
+    gals1 = g1.read_gals()
+    gals2 = g2.read_gals()
+
+    for field in g1.dtype.names:
+        if field in ignored_fields:
             continue
-        
-        t1 = g1.read_tree(treenum)
-        t2 = g2.read_tree(treenum)
-        if (t1 is None and t1 != t2) or \
-           (t2 is None and t1 != t2):
-            msg = "Error: Exactly one of the trees contains "\
-                  "no galaxies.\nt1 = {0}\nt2 = {1}\n"\
-                  .format(t1, t2)
-            raise ValueError(msg)
 
-        xx = np.argsort(t1['SimulationHaloIndex'])
```

Comment:
This is an useful feature to keep around. Perhaps add a boolean keyword that has a default value, and then allow the user to pass in that keyword. The `argsort` is required for cases where **only** the output ordering of galaxies is different